tests/approx_matching_ut.cpp

```diff
@@ -201,50 +201,91 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 		}
 	}
 
+	enum MutateOperation {
+		Substitute = 0,
+		Delete = 1,
+		Insert = 2
+	};
+	
+	ystring ChangeText(const ystring& text, int operation, int posLeft, int posRight = -1)
+	{
+		auto changedText = text;
+		switch (operation) {
+			case MutateOperation::Substitute:
+				if (posRight >= 0) {
```

Comment:
This way operations in both positions are the same.
It would be a little better to make them independent.
Eg.:
```
ystring ChangeText(const ystring& text, int operation, int pos) {
...
}

...
for (int operationLeft = MutateOperation::Begin; operationLeft < OperationCount; ++operationLeft) {
        for (int operationRight = MutateOperation::Begin; operationRight < MutateOperation::End; ++operationRight) {
                auto changedText = ChangeText(text, operationLeft, posLeft);
                changedText = ChangeText(changedText, operationRight, posRight);
        ...
}
```